components/service/glean/src/main/java/mozilla/components/service/glean/ping/BaselinePing.kt

```diff
@@ -69,4 +72,45 @@ internal class BaselinePing(applicationContext: Context) {
             it.id
         }
     }
+
+    /**
+     * Gets a gecko-compatible locale string (e.g. "es-ES" instead of Java [Locale]
+     * "es_ES") for the default locale.
+     *
+     * This method approximates the API21 method [Locale.toLanguageTag].
+     *
+     * @return a locale string that supports custom injected languages.
+     */
+    internal fun getLanguageTag(): String {
+        // Thanks to toLanguageTag() being introduced in API21, we could have
+        // simple returned `locale.toLanguageTag();` from this function. However
+        // what kind of languages the Android build support is up to the manufacturer
+        // and our apps usually support translations for more rare languages, through
+        // our custom locale injector. For this reason, we can't use `toLanguageTag`
+        // and must try to replicate its logic ourselves.
+        val locale = Locale.getDefault()
+        val language = getLanguage(locale)
+        val country = locale.country // Can be an empty string.
+        return if (country.isEmpty()) language else "$language-$country"
```

Comment:
TIL: Linux seems to put an `_` between language and country, but it looks like the [IETF standard](https://tools.ietf.org/html/bcp47#section-2.1.1) is to use hypens, so :+1: to this.